tails_files/install.sh

```diff
@@ -68,23 +87,360 @@ if [ -f $XSESSION_RC ]; then
     # breaking Tor by changing the Sandbox directive while it's
     # running.
     if grep -q 'Sandbox 1' /etc/tor/torrc && ! grep -q 'Sandbox 1' /etc/tor/torrc.bak; then
-        echo "Sandbox 1" >> /etc/tor/torrc.bak
+      echo "Sandbox 1" >> /etc/tor/torrc.bak
     fi
-fi
+  fi
 
-# set up NetworkManager hook
-if ! grep -q 'custom-nm-hooks' "$TAILSCFG/persistence.conf"; then
-  echo "/etc/NetworkManager/dispatcher.d	source=custom-nm-hooks,link" >> $TAILSCFG/persistence.conf
-fi
-mkdir -p $TAILSCFG/custom-nm-hooks
-cp -p $INSTALL_DIR/99-tor-reload.sh $TAILSCFG/custom-nm-hooks
-cp -p $INSTALL_DIR/99-tor-reload.sh /etc/NetworkManager/dispatcher.d/
+  # Remove previous NetworkManager hook if present. In 0.3.6 the prefix
+  # was "99-", which caused it to run too late. In 0.3.5 the prefix was "70-"
+  # and should also be removed for backwards-compatibility. There are a variety
+  # of directories the associated files were placed in, so remove from all locations,
+  # otherwise we'll have multiple redundant hooks. Under Tails v2, if you restart
+  # the tor service too rapidly in succession, it will refuse to start.
+  for d in $tails_live_persistence \
+      $securedrop_dotfiles \
+      $network_manager_dispatcher \
+      "${tails_live_persistence}/custom-nm-hooks"; do
+    for f in 70-tor-reload.sh 99-tor-reload.sh ; do
+      rm -f "${d}/${f}" > /dev/null 2>&1
+    done
+  done
+
+  # Remove binary setuid wrapper from previous tails_files installation, if it exists.
+  rm -f "${securedrop_dotfiles}/securedrop_init" 2>&1
+}
+
+
+function copy_securedrop_dotfiles()
+{
+  mkdir -p "$securedrop_dotfiles"
+
+  # copy icon, launchers and scripts
+  cp -f securedrop_icon.png $securedrop_dotfiles
+  cp -f document.desktop $securedrop_dotfiles
+  cp -f source.desktop $securedrop_dotfiles
+  cp -f securedrop_init.py $securedrop_init_script
+
+  # set permissions
+  chmod 755 $securedrop_dotfiles
+
+}
+
+function set_permissions_on_securedrop_dotfiles()
+{
+  # Follow up function to enforce correct permissions and ownership
+  # on SecureDrop dotfiles in the staging directly. Separated
+  # into function to aid readability.
+  chown root:root $securedrop_init_script
+  chmod 700 $securedrop_init_script
+  chown root:root $torrc_additions
```

Comment:
There's a problem here. Because the previous function `copy_securedrop_dotfiles()` has not copied the `torrc_additions` template into `.securedrop`, we are chown'ing a file that does not exist yet.

`chown: cannot access '/home/amnesia/Persistent/.securedrop/torrc_additions': No such file or directory`

To fix this, you can either cp `torrc_additions` from tails_files in the preceding function, or re-order the program flow so that `configure_torrc_additions()` comes earlier.